lms/views/api/blackboard/files.py

```diff
@@ -14,21 +28,32 @@ def __init__(self, request):
     @view_config(request_method="GET", route_name="blackboard_api.courses.files.list")
     def list_files(self):
         """Return the list of files in the given course."""
-        return self.blackboard_api_client.list_files(
-            self.request.matchdict["course_id"]
+        course_id = self.request.matchdict["course_id"]
+        response = self.blackboard_api_client.request(
+            "GET", f"courses/uuid:{course_id}/resources"
         )
+        return BlackboardListFilesSchema(response).parse()
 
     @view_config(request_method="GET", route_name="blackboard_api.files.via_url")
     def via_url(self):
         """Return the Via URL for annotating the given Blackboard file."""
 
-        return {
-            "via_url": helpers.via_url(
-                self.request,
-                self.blackboard_api_client.public_url(
-                    self.request.matchdict["course_id"],
-                    self.request.params["document_url"],
-                ),
-                content_type="pdf",
+        course_id = self.request.matchdict["course_id"]
+        document_url = self.request.params["document_url"]
+
+        file_id = DOCUMENT_URL_REGEX.search(document_url)["file_id"]
+
+        try:
+            response = self.blackboard_api_client.request(
+                "GET", f"courses/uuid:{course_id}/resources/{file_id}"
             )
-        }
+        except HTTPError as err:
+            if err.response.status_code == 404:
+                raise BlackboardFileNotFoundInCourse(file_id) from err
+            raise
+
+        public_url = BlackboardPublicURLSchema(response).parse()
+
+        via_url = helpers.via_url(self.request, public_url, content_type="pdf")
+
+        return {"via_url": via_url}
```

Comment:
The `via_url()` view is a bit longer but still the only bit of logic is a single `if`. It only takes three unit tests to fully cover this view